libs/stats/odc/stats/tasks.py

```diff
@@ -210,7 +211,15 @@ def msg_default(msg):
         if isinstance(temporal_range, str):
             temporal_range = DateTimeRange(temporal_range)
 
-        if dss is None:
+        if '-' in product:
```

Comment:
If I understand this correctly. It's gathering datasets for multiple output products. I'll assume it's needed in mangroves or it's an attempt to run tcw percentile on fly? Is the component of `product` different from product name we see in the output such that there is no `-` in each one?

Reply:
yep that's right. I've added it here so that `tcw` can use all the landsat products simultaneously. AFAIK the input product names aren't anywhere in the output

Reply:
OK...I got a bit confused then. Is `product` the output product or input product, from so far what I read, it's output product (I could be wrong). I'll assume it's product name intuitively, e.g., in this case `tcw-percentiles`. so when you do `product.split('-')`, we got `tcw` and `percentile`? Can you give me an output string of `product` if possible?

Reply:
all right, I'm gonna believe there is a grammar in naming product so `-` won't be in any product name.